src/render.rs

```diff
@@ -0,0 +1,480 @@
+    render_device: Res<RenderDevice>,
+    pipeline: Res<ParticlePipeline>,
+    textures: Res<RenderAssets<Image>>,
+) {
+    for (entity, mut extracted_instance_data) in particle_system_query.iter_mut() {
+        // Sort the particles only if required by the provided settings
+        if extracted_instance_data.sort_by_depth {
+            // Retrieve the extracted instance data and sort it according to the first given camera (how to know its the main one ?)
+            // WARNING: Here we assume that there is only one ExtractedView at this stage
+            let camera_global_position =
+                extracted_view.get_single().unwrap().transform.translation();
+
+            // Create a Vec with the depth calculated
+            let distances: Vec<f32> = extracted_instance_data
+                .instance_data
+                .iter()
+                .map(|data| -data.position.distance_squared(camera_global_position))
+                .collect();
+
+            // Create a Vec with indices
+            let mut indices: Vec<usize> = (0..distances.len()).collect();
+
+            // Sort the indices Vec according to the distances
+            indices.sort_unstable_by(|&a, &b| {
+                distances[a]
+                    .partial_cmp(&distances[b])
+                    .unwrap_or(Ordering::Less)
+            });
+
+            // We then sort the extracted_instance_data according to the indices Vec
+            for i in 0..indices.len() {
+                while i != indices[i] {
+                    let swap_idx = indices[i];
+                    extracted_instance_data.instance_data.swap(i, swap_idx);
+                    indices.swap(i, swap_idx);
+                }
+            }
+
+            // The standard sort method requires to calculate the square distance twice for each index.
+            /*extracted_instance_data.instance_data.sort_by(|&a, &b|
+            //    b.position.distance_squared(camera_global_position)
+            //        .partial_cmp(&a.position.distance_squared(camera_global_position))
+            //        .unwrap_or(Ordering::Equal)
+            //);*/
+        }
+
+        // Make a buffer out of the extracted instance data
+        let buffer = render_device.create_buffer_with_data(&BufferInitDescriptor {
+            label: Some("instance data buffer"),
+            contents: { bytemuck::cast_slice(extracted_instance_data.instance_data.as_slice()) },
+            usage: BufferUsages::VERTEX | BufferUsages::COPY_DST,
+        });
+
+        // If no texture was provided, use the dummy texture of the mesh pipeline `[MeshPipeline::dummy_white_gpu_image]`
+        let my_texture = if let Some(tex) = &extracted_instance_data.texture {
+            textures.get(tex).unwrap()
+        } else {
+            &pipeline.mesh_pipeline.dummy_white_gpu_image
+        };
+
+        // Create the bind group for the particle system
+        let ps_bind_group = render_device.create_bind_group(&BindGroupDescriptor {
```

Comment:
Same idea here, I think we can avoid creating these every frame.

Reply:
I should check how RenderAssets behave, but I think we will still need to update the `BindingResource` each frame. But we could try to store the bind group as a Resource for each particle system yes !